docs/content/040-adoption-guides/040-nexus-with-nextjs.md

```diff
@@ -0,0 +1,328 @@
+```
+
+Now in the `graphql.ts` for our API route, we can simple pull in the constructed schema and serve it with Apollo Server.
+
+```ts
+// pages/api/graphql.ts
+
+import { ApolloServer } from 'apollo-server-micro'
+import schema from './../../schema'
+
+const server = new ApolloServer({
+  schema,
+})
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+}
+
+export default server.createHandler({
+  path: '/api/graphql',
+})
+```
+
+If we head back over to `localhost:3000/api/graphql`, we can see results for our call for `frameworks`.
+
+![graphql playground showing a list of front end frameworks from a graphql query](/assets/next-nexus-2.png)
+
+## Call the GraphQL API from the Front End
+
+In a Next.js app, we have several options for producing our pages:
+
+- Server-side rendering using the `getServerSideProps` function
+- Static site generation using the `getStaticProps` function
+- Client-side rendering
+
+No matter how we want to produce our pages, we can call our API to get data to display.
+
+If we would like to use server-side rendering, we can use a library like `graphql-request` to make an API call.
+
+Start by intalling the `graphql-request` library.
+
+```bash
+npm i graphql-request
+```
+
+Create a function called `getServerSideProps` and make a call to our GraphQL API. The returned data can be supplied in the returned object which will make it available as a `prop` in the function that produces the page. Next.js will take care of treating this page as a server-side rendered page.
+
+```ts
+// pages/index.tsx
+
+import { request, gql } from 'graphql-request'
+
+export async function getServerSideProps() {
+  const query = gql`
+    {
+      frameworks {
+        id
+        name
+      }
+    }
+  `
+  const data = await request('http://localhost:3000/api/graphql', query)
+  const { frameworks } = data
+  return {
+    props: {
+      frameworks,
+    },
+  }
+}
+
+export default function Home({ frameworks }) {
+  return (
+    <div>
+      <ul>
+        {frameworks.map((f) => (
+          <li key={f.id}>{f.name}</li>
+        ))}
+      </ul>
+    </div>
+  )
+}
+```
+
+When the page renders, we'll see a list of the frameworks.
+
+![a next.js page rendered with a list of frameworks](/assets/next-nexus-3.png)
+
+**Note:** Using `graphql-request` to make a server-side call for data from our GraphQL API is good for initially fetching the data but it won't do much for us if we want to have client-side GraphQL features such as caching. If we want additional features like those provided by Apollo Client, it won't work to mix the data retrieved in the `getServerSideProps` function with data managed on the client by Apollo.
+
+## Schema and Type Generation
+
+A GraphQL SDL file and Nexus typings can automatically be generated by configuring the `outputs` option in `makeSchema`. In a typical project, we can tell Nexus where to place these files by using `__dirname` to define a path. That might looks something like this:
```

Comment:
Since these files will be generated by default soon. So something like this?

```suggestion
Nexus generates a GraphQL SDL file and TypeScript typings by default. You can customize their location by configuring `outputs` option in `makeSchema`. In a typical project, we can tell Nexus where to place these files by using `__dirname` to define a path. That might looks something like this:
```